### DISCUSSION TITLE
TCP : Unreachable peer Error

### DISCUSSION THREAD

**Discussion by BeyzaOzkara** [gijzelaerr/python-snap7#485] [Q&A] [answered]
I am getting a "TCP: Unreachable peer" error intermittently when attempting to read data from the PLC. My script generally works fine, but this error occasionally occurs, impacting the reliability of the data retrieval process.

- I've verified the PLC's connectivity and ensured that it is powered on.
- The IP address and port in the code seem to be correctly configured.
- The error suggests a network-related issue, but I'm unable to pinpoint the exact cause.

 I would appreciate any insights or suggestions you may have.

**Comment by spreeker** [comment, marked as answer]
HI,

Periodically completely rebuild the connection. Lost network connections are common and your code should be able to recover from that.

**Comment by BeyzaOzkara** [comment]
Hi, 
Thank you for the prompt response. I wasn't sure if this was a normal problem to have. Thanks again for clarifying it.
